addons/stock/wizard/stock_quant_relocate.py

```diff
@@ -0,0 +1,72 @@
+    message = fields.Text('Reason for relocation')
+    is_not_full_package = fields.Boolean(compute='_compute_is_not_full_package')
+    partial_package_names = fields.Char(compute="_compute_is_not_full_package")
+    is_not_single_location = fields.Boolean(compute='_compute_is_not_single_location')
+
+    @api.depends('quant_ids')
+    def _compute_is_not_full_package(self):
+        for wizard in self:
+            wizard.is_not_full_package = False
+            wizard.partial_package_names = ''
+            packages = wizard.quant_ids.package_id
+            incomplete_packages = packages.filtered(lambda p: any(q not in wizard.quant_ids.ids for q in p.quant_ids.ids))
+            if packages and incomplete_packages:
+                wizard.is_not_full_package = True
+                wizard.partial_package_names = ', '.join(incomplete_packages.mapped('display_name'))
+
+    @api.depends('dest_location_id', 'quant_ids')
+    def _compute_is_not_single_location(self):
+        for wizard in self:
+            wizard.is_not_single_location = False
+            if len(wizard.quant_ids.location_id) > 1 and not wizard.dest_location_id:
+                wizard.is_not_single_location = True
+
+    @api.depends('dest_location_id', 'quant_ids')
+    def _compute_dest_package_id_domain(self):
+        for wizard in self:
+            domain = ['|', ('company_id', '=', wizard.company_id.id), ('company_id', '=', False)]
+            if wizard.dest_location_id:
+                domain += ['|', ('location_id', '=', False), ('location_id', '=', wizard.dest_location_id.id)]
+            elif len(wizard.quant_ids.location_id) == 1:
+                domain += ['|', ('location_id', '=', False), ('location_id', '=', wizard.quant_ids.location_id.id)]
+            wizard.dest_package_id_domain = domain
+
+    @api.depends('dest_package_id_domain')
+    def _compute_dest_package_id(self):
+        for wizard in self:
+            if wizard.dest_package_id and wizard.dest_package_id not in wizard.dest_package_id.search(literal_eval(wizard.dest_package_id_domain)):
+                wizard.dest_package_id = False
+
+    def action_relocate_quants(self):
+        self.ensure_one()
+        if not self.dest_location_id and not self.dest_package_id:
+            return
+        self.quant_ids.action_clear_inventory_quantity()
+
+        if self.is_not_full_package and not self.dest_package_id:
+            quants_to_unpack = self.quant_ids.filtered(lambda q: not all(sub_q in self.quant_ids.ids for sub_q in q.package_id.quant_ids.ids))
+            quants_to_unpack.move_quants(location_dest_id=self.dest_location_id, message=self.message, unpack=True)
+            self.quant_ids -= quants_to_unpack
+        self.quant_ids.move_quants(location_dest_id=self.dest_location_id, package_dest_id=self.dest_package_id, message=self.message)
+
+        if self.env.context['default_lot_id']:
```

Comment:
just in case
```suggestion
        if self.env.context.get('default_lot_id', false) and len(self.quant_ids.lot_id) == 1:
```

Should also maybe be tested to make sure the `default_lot_id` isn't too sticky and causes unexpected behavior when clicking through the views